lang/csharp/src/apache/main/Schema/ArraySchema.cs

```diff
@@ -48,11 +48,12 @@ internal static ArraySchema NewInstance(JToken jtok, PropertyMap props, SchemaNa
         }
 
         /// <summary>
-        /// Constructor
+        /// Initializes a new instance of the <see cref="ArraySchema"/> class.
         /// </summary>
-        /// <param name="items">schema for the array items type</param>
-        /// <param name="props">dictionary that provides access to custom properties</param>
-        private ArraySchema(Schema items, PropertyMap props) : base(Type.Array, props)
+        /// <param name="items">schema for the array items type.</param>
+        /// <param name="customAttributes">dictionary that provides access to custom properties.</param>
+        public ArraySchema(Schema items, PropertyMap customAttributes = null)
```

Comment:
Schema classes already use a Factory pattern method "NewInstance".  I would recommend following the same pattern than exposing the constructor.